lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -1185,7 +1149,7 @@ class SemanticsObject {
   bool _isDirty(int fieldIndex) => (_dirtyFields & fieldIndex) != 0;
 
   /// The dom element of this semantics object.
-  DomElement get element => primaryRole!.element;
+  DomElement get element => semanticRole!.element;
```

Comment:
The class `SemanticObject` looks a little bit too complicated, shouldn't each `SemanticRole`/`Behavior` know how to update and identify itself? Is this class doing too much, does it need love? Is it needed?

Reply:
`SemanticRole` can change, but a `SemanticObject` is persistent for the entire lifecycle of the node. So for now we need both. However, we can revisit who's responsible for what, since both objects are 1:1 with the node. For now, I'm keeping this PR to non-functional changes only, just renames and deletion of dead code.